lib/ocpp/v201/average_meter_values.cpp

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2020 - 2023 Pionix GmbH and Contributors to EVerest
+
+#include <everest/logging.hpp>
+#include <everest/timer.hpp>
+#include <ocpp/v201/average_meter_values.hpp>
+
+namespace ocpp {
+namespace v201 {
+AverageMeterValues::AverageMeterValues() {
+}
+void AverageMeterValues::clear_values() {
+    this->aligned_meter_values.clear();
+    this->averaged_meter_values.sampledValue.clear();
+}
+
+void AverageMeterValues::set_values(const MeterValue& meter_value) {
+    std::lock_guard<std::mutex> lk(this->avg_meter_value_mutex);
+    // store all the meter values in the struct
+    this->averaged_meter_values = meter_value;
+
+    // avg all the possible measurerands
+    for (auto element : meter_value.sampledValue) {
+        if (is_avg_meas(element)) {
+            MeterValueCalc temp =
```

Comment:
If you make this a reference you can edit the value inside the map instead of having to copy it twice